install/installer.class.php

```diff
@@ -494,20 +464,74 @@ public function createNotificationTargetInvitation() {
     *
     * @param string $fromVersion
     */
-   protected function upgrade($fromVersion) {
-      switch ($fromVersion) {
-         case '2.0.0':
-            // Example : upgrade to version 3.0.0
-            // $this->upgradeOneStep('3.0.0');
-         case '3.0.0':
-            // Example : upgrade to version 4.0.0
-            // $this->upgradeOneStep('4.0.0');
+   protected function upgrade(Migration $migration) {
+      $this->migration = $migration;
+      $fromSchemaVersion = $this->getSchemaVersion();
+
+      switch ($fromSchemaVersion) {
+         case '2.0':
+            // Example : upgrade to schema version 3.0
+            // $this->upgradeOneStep('3.0');
+         case '3.0':
+            // Example : upgrade to version 4.0
+            // $this->upgradeOneStep('4.0');
 
          default:
       }
       if (PluginFlyvemdmCommon::endsWith(PLUGIN_FLYVEMDM_VERSION, "-dev")) {
          $this->upgradeOneStep('dev');
       }
+
+      $this->migration->executeMigration();
+
+      $this->createDirectories();
+      $this->createFirstAccess();
+      $this->createGuestProfileAccess();
+      $this->createAgentProfileAccess();
+      $this->createDefaultFleet();
+      $this->createPolicies();
+      $this->createNotificationTargetInvitation();
+      $this->createJobs();
+      $this->createRootEntityConfig();
+      $this->createDisplayPreferences();
+
+      Config::setConfigurationValues(
+         'flyvemdm', [
+            'version' => PLUGIN_FLYVEMDM_VERSION,
+            'schema_version' => PLUGIN_FLYVEMDM_SCHEMA_VERSION,
+         ]
+      );
+   }
+
+   /**
+    * Find the version of the plugin
+    *
+    * @return string|NULL
+    */
+   protected function getSchemaVersion() {
+      return $this->getSchemaVersionFromGlpiConfig();
+   }
```

Comment:
this method might be useful to handle schema version detection without relying on the data in the configuration (when we find a bug lately).